File: backend-old/app/models/batch.py
from pydantic import BaseModel
from typing import List

class BatchRequest(BaseModel):
    total_transactions: int
    total_amount: float
    duration_seconds: int
    merchant_id: str

class BatchResponse(BaseModel):
    success_count: int
    failure_count: int
    average_response_time: float
    total_processed_amount: float
    transactions: List[dict]

File: backend/src/models/Batch.js
const Joi = require('joi');

// Batch request validation schema
const batchRequestSchema = Joi.object({
  total_transactions: Joi.number()
    .integer()
    .positive()
    .required(),
  
  total_amount: Joi.number()
    .positive()
    .required(),
  
  duration_seconds: Joi.number()
    .integer()
    .positive()
    .required(),
  
  merchant_id: Joi.string()
    .required()
});

// Batch response structure
class BatchResponse {
  constructor({ success_count, failure_count, average_response_time, total_processed_amount, transactions }) {
    this.success_count = success_count;
    this.failure_count = failure_count;
    this.average_response_time = average_response_time;
    this.total_processed_amount = total_processed_amount;
    this.transactions = transactions;
  }
}

module.exports = {
  batchRequestSchema,
  BatchResponse,
  validateBatchRequest: (data) => batchRequestSchema.validate(data)
};